[TYPING] pandas.io.formats.format._binify annotation is inconsistent with  DataFrameFormatter.__init__

#### Problem description

`pd.io.formats.format._binify` defines `line_width` as `Union[np.int32, int]`:

https://github.com/pandas-dev/pandas/blob/39602e7d7c5b663696f5a6eca9e66e65483bc868/pandas/io/formats/format.py#L1893

however it is inconsistent with `DataFrameFormatter.__init__` 

https://github.com/pandas-dev/pandas/blob/39602e7d7c5b663696f5a6eca9e66e65483bc868/pandas/io/formats/format.py#L518

and `DataFrameFormatter._join_multiline`

https://github.com/pandas-dev/pandas/blob/39602e7d7c5b663696f5a6eca9e66e65483bc868/pandas/io/formats/format.py#L871

which take and pass as-is `Optional[int]`.

This leads to Mypy error:

```
pandas/io/formats/format.py:871: error: Argument 2 to "_binify" has incompatible type "Optional[int]"; expected "Union[int32, int]"
```

Providing default in [`_join_multiline`](https://github.com/pandas-dev/pandas/blob/39602e7d7c5b663696f5a6eca9e66e65483bc868/pandas/io/formats/format.py#L860), i.e.

```python
lwidth = self.line_width or DEFAULT_LINE_WIDTH
```
could be reasonable fix. One could also ignore [the call](https://github.com/pandas-dev/pandas/blob/39602e7d7c5b663696f5a6eca9e66e65483bc868/pandas/io/formats/format.py#L871):

```python
col_bins = _binify(col_widths, lwidth)  # type: ignore
```

but that leaves possible correctness problem.



#### Expected Output

Passing Mypy type check

#### Output of ``pd.show_versions()``

```
INSTALLED VERSIONS
------------------
commit           : bee17d5f4c99e6d1e451cf9a7b6a1780aa25988d
python           : 3.7.3.final.0
...
```
Additionally
```
mypy==0.730
```

CC @WillAyd

changing to `def _binify(cols: List[int], line_width: int) -> List[int]:`

and I get `pandas\io\formats\format.py:873:40: error: Argument 2 to "_binify" has incompatible type "Optional[int]"; expected "int"  [arg-type]`

so the Any is swallowing the None.

@zero323 do you have numpy stubs in your environment?

--- a/pandas/io/formats/format.py
+++ b/pandas/io/formats/format.py
@@ -868,6 +868,8 @@ class DataFrameFormatter(TableFormatter):
             np.array([self.adj.len(x) for x in col]).max() if len(col) > 0 else 0
             for col in strcols
         ]
+
+        assert lwidth is not None
         col_bins = _binify(col_widths, lwidth)
         nbins = len(col_bins)
 
@@ -1890,7 +1892,7 @@ def set_eng_float_format(accuracy: int = 3, use_eng_prefix: bool = False) -> Non
     set_option("display.column_space", max(12, accuracy + 9))
 
 
-def _binify(cols: List[np.int32], line_width: Union[np.int32, int]) -> List[int]:
+def _binify(cols: List[int], line_width: int) -> List[int]:
     adjoin_width = 1
     bins = []
     curr_width = 0

